src/main/java/com/marklogic/kafka/connect/DefaultDatabaseClientConfigBuilder.java

```diff
@@ -35,10 +82,98 @@ public DatabaseClientConfig buildDatabaseClientConfig(Map<String, String> kafkaC
 
 		String securityContextType = kafkaConfig.get(MarkLogicSinkConfig.CONNECTION_SECURITY_CONTEXT_TYPE).toUpperCase();
 		clientConfig.setSecurityContextType(SecurityContextType.valueOf(securityContextType));
-
+		/*
+		 * v1.2.2 changes START
+		 */		
+		if ("BASIC".equals(securityContextType) ||
+				"DIGEST".equals(securityContextType)
+				) {
+					if (ssl != null && Boolean.parseBoolean(ssl)) {
+						if (sslMutualAuth != null && Boolean.parseBoolean(sslMutualAuth)) {
+							/*2 way ssl changes*/
+							KeyStore clientKeyStore = null;
+							try {
+								clientKeyStore = KeyStore.getInstance("PKCS12");
+							} catch (KeyStoreException e) {
+								
+								throw new RuntimeException("Unable to get default SSLContext: " + e.getMessage(), e);
+							}
+					        TrustManager[] trust = new TrustManager[] { new SimpleX509TrustManager()};
+							
+					        try (InputStream keystoreInputStream = new FileInputStream(clientConfig.getCertFile())) {
+					            clientKeyStore.load(keystoreInputStream, clientConfig.getCertPassword().toCharArray());
+					        } catch (FileNotFoundException e) {
```

Comment:
Since you have the same logic for handling each exception type, just catch Exception and throw a RuntimeException. Also, the message isn't accurate - you're not trying to get the default SSLContext, you're trying to load a configure a custom key store. I would have a general message of "Unable to configure custom SSL connection: " + e.getMessage().

Reply:
Making the changes